src/i_video.c

```diff
@@ -494,28 +484,43 @@ static void DynamicResolution(void)
     double actualpercent = actual / target;
 
     #define DRS_MIN_HEIGHT 400
-    #define DRS_DELTA 0.5
+    #define DRS_DELTA 0.01
     #define DRS_GREATER (1 + DRS_DELTA)
     #define DRS_LESS (1 - DRS_DELTA)
+    #define DRS_STEP (SCREENHEIGHT / 2)
 
     int newheight = 0;
     int oldheight = video.height;
 
+    frame_counter++;
+    averagepercent = (averagepercent + actualpercent) / frame_counter;
+
     if (actualpercent > DRS_GREATER)
     {
         double reduction = (actualpercent - DRS_GREATER ) * 0.2;
         newheight = (int)MAX(DRS_MIN_HEIGHT, oldheight - oldheight * reduction);
     }
-    else if (actualpercent < DRS_LESS)
+    else if (averagepercent < DRS_LESS && frame_counter > targetrefresh)
```

Comment:
@bangstk Now it increases the resolution after about 1 second. A simple approach, but it works well enough for me.